tool/client.cc

```diff
@@ -245,19 +245,89 @@ static bool WaitForSession(SSL *ssl, int sock) {
   return true;
 }
 
+static void print_verify_details(SSL *s) {
+  long verify_err = SSL_get_verify_result(s);
+
+  if (verify_err == X509_V_OK) {
+    fprintf(stdout, "Verification: OK\n");
+  } else {
+    const char *reason = X509_verify_cert_error_string(verify_err);
+    fprintf(stdout, "Verification error: %s\n", reason);
+  }
+}
+
+static void PrintOpenSSLConnectionInfo(SSL *ssl, int show_certs) {
+  STACK_OF(X509) *sk;
+  X509 *peer;
+  int i;
+
+  sk = SSL_get_peer_cert_chain(ssl);
+  if (sk != NULL) {
+    fprintf(stdout, "---\nCertificate chain\n");
+    for (i = 0; i < (int) sk_X509_num(sk); i++) {
+      fprintf(stdout, "%2d s:", i);
+      X509_NAME_print_ex_fp(stdout, X509_get_subject_name(sk_X509_value(sk, i)),
+                            0, XN_FLAG_ONELINE);
+      fprintf(stdout, "\n   i:");
+      X509_NAME_print_ex_fp(stdout, X509_get_issuer_name(sk_X509_value(sk, i)),
+                            0, XN_FLAG_ONELINE);
+      fprintf(stdout, "\n");
+      if (show_certs) {
+        PEM_write_X509(stdout, sk_X509_value(sk, i));
+      }
+    }
+  }
+
+  fprintf(stdout, "---\n");
+  peer = SSL_get_peer_certificate(ssl);
+  if (peer != NULL) {
+    fprintf(stdout, "Server certificate\n");
+    PEM_write_X509(stdout, peer);
+    fprintf(stdout, "subject=");
+    X509_NAME_print_ex_fp(stdout,
+                       X509_get_subject_name(peer),
+                       0, XN_FLAG_ONELINE);
+    fprintf(stdout, "\n\nissuer=");
+    X509_NAME_print_ex_fp(stdout,
+                          X509_get_issuer_name(peer),
+                          0, XN_FLAG_ONELINE);
+    fprintf(stdout, "\n\n---\n");
+  } else {
+    fprintf(stdout, "no peer certificate available\n");
+  }
+
+  // TODO (aws-lc): we are missing some functions needed to print the following data
+  //  print_ca_names(bio, s);
+  //  ssl_print_sigalgs(bio, s);
+  //  ssl_print_tmp_key(bio, s);
+
+  fprintf(stdout,
+             "---\nSSL handshake has read %d bytes "
+             "and written %d bytes\n",
+        (int)BIO_number_read(SSL_get_rbio(ssl)),
+        (int)BIO_number_written(SSL_get_wbio(ssl)));
+  print_verify_details(ssl);
+  X509_free(peer);
+}
+
 static bool DoConnection(SSL_CTX *ctx,
                          std::map<std::string, std::string> args_map,
-                         bool (*cb)(SSL *ssl, int sock)) {
+                         bool (*cb)(SSL *ssl, int sock), int tool) {
```

Comment:
Same boolean comment here, but I think we could also rename the parameter name to be more clearer. Probably could also consider using `is_openssl_s_client` here